docs/src/pages/ComponentArchitecture.mdx

```diff
@@ -0,0 +1,97 @@
+
+Hooks should be created to manage internal state of base components. For sets
+of functionality common to several components, shared hooks should be abstracted
+and exported from a common platform package.
+
+#### There is a limited set of accessibility options that work well together, and each combination should be tested cohesively. Further, accessibility is closely tied to a components internal state, and they are likely to iterate concurrently.
+
+Each base component with accessibility concerns should provide its own implementation of accessibility props and handlers.
+For variants that are expected in the overall component library, state hooks should be reused and a new component simply
+created with a different use-case.
+
+It is important to have consistent accessibility, and passing in a new accessibility model might create major
+challenges to keep our components accessible, and also predictable. If a particular component needs to have multiple
+accessibility models (keyboarding, attributes), it should make that decision based on a single prop with a fixed set
+of implementation options (i.e. a `accessibilityMode` prop that accepts a fixed set of string values as input).
+
+# Themed components
+
+The control library should ship a set of unopinionated components (i.e. base components) but also components which are
+pre-scaffolded with Fluent design opinions (themed components). Fluent will be opinionated but customizable.
+
+#### Themed components are the glue that ties together context, theming, and the base component. There are several potential implementations (`createComponent`, other HoCs), but serve as the higher level abstraction that brings only the necessary pieces to a consumer's application bundle.
+
+Themed components should pass a set of classes to base components. In practice, this will mean pulling
+together information about theming and styling from context, and ensuring the correct CSS is rendered on
+the page.
+
+#### The themed layer is the preferred layer to add hard graph edges. Each themed control should be intuitive to use for a developer, and should come with "batteries included".
+
+Each themed component that relies on sub-components should specify which components to render through
+the slots prop.
+
+The slots prop must be accessible in several means to ensure that a "recursive component" can be accurately described.
+
+#### We want to offer a delightful first-run experience, and bundle size will not be negatively impacted by including the default theme.
+
+Each component should render a known-good theme out of the box when used without a provider.
+
+#### As a component library cannot anticipate all needs to applications' design systems, allowances should be made to extend the design system while maintaining rich levels of type safety.
+
+The HoC used to glue base components to the rest of the system should also provide a facility
```

Comment:
If HoC referres to Higher order component, can we avoid it, as it implies details in implementation that we haven't decided on. I would really be agains composing components and polluting the react tree.